### PR TITLE
Google lighthouse performance score audit for mobile

### PR DESCRIPTION
**Pull request by limsohee1002**
Description
 - Add dynamic import
 - Add smaller device size for image
 - Add minimum cache ttl: https://nextjs.org/docs/api-reference/next/image#minimum-cache-ttl
 - Add revalidate
 - Use next script 

Desktop:
image

Mobile:
image

### CODE REVIEW

**Comment by limsohee1002** [comment]
image

Hey @hhff  There are a few things that have left to do and I have a few questions:
- One most confusing thing is, the score changes whenever I run the test even if I didn't change anything. It is kind of hard to figure out what is a real score. 
-  Enable text compression:
image
Google wants the font file to be compressed. I guess I am not too sure how I can do that. Can I just gzip the file and re-upload it to s3? It should be really fast so I am not too sure this would be necessary either.
- Minimize main thread work
image
There are things that google recommended to do (https://web.dev/mainthread-work-breakdown/?utm_source=lighthouse&utm_medium=devtools) 
Things that I am thinking that might help are: 
    - remove 3rd party libraries and implement the function without it. 
    - go through the CSS files and remove complex calculations - complex calculations might be in basement CSS. I will introduce purge css

What do you think?

**Comment by hhff** [comment]
> Google wants the font file to be compressed. I guess I am not too sure how I can do that. Can I just gzip the file and re-upload it to s3? It should be really fast so I am not too sure this would be necessary either.

@limsohee1002 - I just enabled "Compress Objects Automatically" ([read here](https://aws.amazon.com/blogs/aws/new-gzip-compression-support-for-amazon-cloudfront/)) on AWS Cloudfront for serving `style.css`. Can you check that warning is fixed?

**Comment by limsohee1002** [comment]
> > Google wants the font file to be compressed. I guess I am not too sure how I can do that. Can I just gzip the file and re-upload it to s3? It should be really fast so I am not too sure this would be necessary either.
> 
> @limsohee1002 - I just enabled "Compress Objects Automatically" ([read here](https://aws.amazon.com/blogs/aws/new-gzip-compression-support-for-amazon-cloudfront/)) on AWS Cloudfront for serving `style.css`. Can you check that warning is fixed?

It is gone! only one left!
image

**Comment by hhff** [comment]
@limsohee1002 - looks like purgecss deleted the margins on this page:
image

File: styles/global.scss [hunk 1]
```diff
@@ -21,10 +21,6 @@ html {
   transition: background-color 3500ms cubic-bezier(0.79, 0.1, 0.56, 0.93);
 }
 
-.pace .pace-progress {
-  background: black !important;
-}
-
 .debug {
   border: solid blue;
```
Comment (hhff) [inline styles/global.scss:25]: is this class used?

**Comment by limsohee1002** [comment]
image

**Pull request merged**